### PR TITLE
Add treat migration guide

### PR DESCRIPTION
**Pull request by markdalgleish**

### CODE REVIEW

File: docs/treat-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+🎉 First of all, thanks for using treat, and thanks for your interest in vanilla-extract!
+
+When we first started work on vanilla-extract, it actually began life as `treat-next` — an experiment with replacing our custom webpack-based theming system with CSS Variables. However, we quickly realised that the theming system was too different to carry the same name. That said, they're similar enough that migrating shouldn't be too hard (relative to migrating to another library).
+
+We've made sure that both treat and vanilla-extract can run simultaneously in the same project, so if you have a lot of treat files, don't feel like you have to migrate everything at once!
+
+## New file extension
+
+The file extension has changed from `*.treat.ts` to `*.css.ts`.
```
Comment (mattcompiles) [inline docs/treat-migration-guide.md:11]: Should we say default file extension? Also, mention `.js` as well or is that implied?

File: docs/treat-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+```tsx
+// App.tsx
+
+import { createContext, useContext } from 'react';
+import { themeClass } from './vars.css';
+import { VanillaThemeProvider } from './VanillaThemeContext';
+
+export const App = () => (
+  <VanillaThemeContext value={themeClass}>
```
Comment (mattcompiles) [inline docs/treat-migration-guide.md:134]:
```suggestion
  <VanillaThemeProvider value={themeClass}>
```

File: docs/treat-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+    <div className={themeClass}>...</div>
+  </VanillaThemeContext>
```
Comment (mattcompiles) [inline docs/treat-migration-guide.md:136]:
```suggestion
  </VanillaThemeProvider>
```

File: docs/treat-migration-guide.md (hunk 1 above)
Comment (markdalgleish) [reply docs/treat-migration-guide.md:11]: I feel like it's implied—happy to leave it?

**Review by mattcompiles — APPROVED** [review]

**Review by mattcompiles — APPROVED** [review]

**Pull request merged**